### PR TITLE
ci/gha: add all-done jobs

### PR DESCRIPTION
**Pull request by kolyshkin**
The sole reason is to simplify branch protection rules, requiring just these to be passed.

~~Currently a draft, pending #4374 #4360 merge.~~

### CODE REVIEW

**Comment by kolyshkin** [comment]
Once this is merged, we can edit branch protection rules to mark only "ci / all-done" and "validate / all-done" tasks as required, removing all other GHA entries. This solves the issue of needing to edit branch protection rules when we switch Go versions or some such.

**Comment by thaJeztah** [comment]
Ah! I was considering something like this for auto-merge as well; auto-merge has the very confusing behavior to _ONLY_ wait for required checks to complete. Any other check is simply ignored, which means that it can merge automatically if checks fail, or even didn't run at all 🙈 

I wish GitHub actions had some option for this though that didn't require manually constructing the "needs" list 😞

**Review by thaJeztah — APPROVED** [review]
conceptually "LGTM"


happy to learn if someone would know a different approach to do the same that doesn't require manually maintaining the list (@crazy-max do you have some trick up your sleeve?)

**Comment by kolyshkin** [comment]
> happy to learn if someone would know a different approach to do the same that doesn't require manually maintaining the list

Yeah, I thought about that, too, and read 
[this](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs) document twice, alas, couldn't find a way.

**Comment by kolyshkin** [comment]
Can we please merge this @opencontainers/runc-maintainers ? I really want to simplify the branch protection rules.

**Review by AkihiroSuda — APPROVED** [review]

**Comment by kolyshkin** [comment]
Awesome, I've simplified branch protection rules for the main branch, now we only require all-done jobs (and the ones from cirrus-ci).

**Pull request merged**
